validator/block_validator.go

```diff
@@ -740,6 +741,44 @@ func (v *BlockValidator) reorgToBlockImpl(blockNum uint64, blockHash common.Hash
 	return nil
 }
 
+// Must be called after SetCurrentWasmModuleRoot sets the current one
+func (v *BlockValidator) Initialize() error {
+	switch v.config.CurrentModuleRoot {
+	case "latest":
+		latest, err := v.MachineLoader.GetConfig().ReadLatestWasmModuleRoot()
+		if err != nil {
+			return err
+		}
+		v.currentWasmModuleRoot = latest
+	case "current":
+		if (v.currentWasmModuleRoot == common.Hash{}) {
+			return errors.New("wasmModuleRoot set to 'current' - but info not set from chain")
+		}
+	default:
+		v.currentWasmModuleRoot = common.HexToHash(v.config.CurrentModuleRoot)
+	}
+	if err := v.MachineLoader.CreateMachine(v.currentWasmModuleRoot, true); err != nil {
+		return err
+	}
+
+	if v.config.PendingUpgradeModuleRoot != "" {
+		if v.config.PendingUpgradeModuleRoot == "latest" {
+			latest, err := v.MachineLoader.GetConfig().ReadLatestWasmModuleRoot()
+			if err != nil {
+				return err
+			}
+			v.pendingWasmModuleRoot = latest
+		} else {
+			v.pendingWasmModuleRoot = common.HexToHash(v.config.PendingUpgradeModuleRoot)
```

Comment:
Same here - if you put e.g. "current" as the pending config, I think right now it'd produce the zero hash, which would mess things up